compiler/qsc_frontend/src/typeck/tests.rs

```diff
@@ -3189,22 +3189,21 @@ fn lambda_inner_return_without_call_ambiguous() {
             }
         "},
         "",
-        &expect![[r#"
+        &expect![[r##"
             #6 30-32 "()" : Unit
             #10 40-112 "{\n        let f = (a, b) -> {\n            return a + b;\n        };\n    }" : Unit
-            #12 54-55 "f" : ((?2, ?2) -> ?2)
-            #14 58-105 "(a, b) -> {\n            return a + b;\n        }" : ((?2, ?2) -> ?2)
-            #15 58-64 "(a, b)" : (?2, ?2)
-            #16 59-60 "a" : ?2
-            #18 62-63 "b" : ?2
+            #12 54-55 "f" : ((Unit, Unit) -> Unit)
+            #14 58-105 "(a, b) -> {\n            return a + b;\n        }" : ((Unit, Unit) -> Unit)
+            #15 58-64 "(a, b)" : (Unit, Unit)
+            #16 59-60 "a" : Unit
+            #18 62-63 "b" : Unit
             #20 68-105 "{\n            return a + b;\n        }" : Unit
             #21 68-105 "{\n            return a + b;\n        }" : Unit
             #23 82-94 "return a + b" : Unit
-            #24 89-94 "a + b" : ?2
-            #25 89-90 "a" : ?2
-            #28 93-94 "b" : ?2
-            Error(Type(Error(AmbiguousTy(Span { lo: 62, hi: 63 }))))
-        "#]],
+            #24 89-94 "a + b" : Unit
```

Comment:
So, this is no longer an error?

Reply:
Ok, this was a subtle side effect of the way that equality constraints are resolved in the type system. I fixed it and now the test is largely back to what it was, with the one change being the (intentional and correct) treatment of the lambda block as the same type as the inference variable for the return statement. A later test now changes slightly because the number of inference variables is different (goes from `?2` to `?3`) but that is safe. Thanks again for drawing attention to this one!